docs/design/proposals/clusterid-mapping.md

```diff
@@ -0,0 +1,140 @@
+# Design to handle clusterID and poolID for DR
+
+`0001-0009-rook-ceph-0000000000000002-b0285c97-a0ce-11eb-8c66-0242ac110002`
+
+The Above is the sample volumeID sent back in response to the CreateVolume operation
+and added as a volumeHandle in the PV spec. CO (Kubernetes) uses above as the
+identifier for other operations on the volume/PVC.
+
+The VolumeID is encoded as,
+
+```text
+0001 -->                              [csi_id_version=1:4byte] + [-:1byte]
+0009 -->                              [length of clusterID=1:4byte] + [-:1byte]
+rook-ceph -->                         [clusterID:36bytes (MAX)] + [-:1byte]
+0000000000000002 -->                  [poolID:16bytes] + [-:1byte]
+b0285c97-a0ce-11eb-8c66-0242ac110002 --> [ObjectUUID:36bytes]
+Total of constant field lengths, including '-' field separators would hence be,
+4+1+4+1+1+16+1+36 = 64
+```
+
+When mirroring is enabled volume which is `csi-vol-ObjectUUID` is mirrored to
+the other cluster.
+
+> `csi-vol` is const name and over has the option to override it in
+> storageclass.
+
+During the Disaster Recovery (failover operation) the PVC and PV will be
+recreated on the other cluster. When Ceph-CSI receives the request for
+operations like (NodeStage, ExpandVolume, DeleteVolume, etc) the volumeID is
+sent in the request which will help to identify the volume.
+
+```yaml=
+apiVersion: v1
+kind: ConfigMap
+data:
+  config.json: |-
+    [
+      {
+       "clusterID": "rook-ceph",
+        "radosNamespace": "<rados-namespace>",
+        "monitors": [
+          "192.168.39.82:6789"
+        ],
+        "cephFS": {
+          "subvolumeGroup": "<subvolumegroup for cephfs volumes>"
+        }
+      },
+            {
+       "clusterID": "fs-id",
+        "radosNamespace": "<rados-namespace>",
+        "monitors": [
+          "192.168.39.83:6789"
+        ],
+        "cephFS": {
+          "subvolumeGroup": "<subvolumegroup for cephfs volumes>"
+        }
+      }
+    ]
+metadata:
+  name: ceph-csi-config
+```
+
+During CSI/Replication operations, Ceph-CSI will decode the volumeID and gets
+the monitor configuration from the configmap and the poolID it will get the
+pool Name and retrieves the OMAP data stored in the rados OMAP and finally
+check the volume is present in the pool.
+
+We have two problems!
```

Comment:
make this a sub-title, and phrase it more like "Two Problems with volumeID Replication" or so